### PR TITLE
Create ROADMAP.md

### PR DESCRIPTION
**Pull request by keithchambers**
List of existing and roadmap features.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+1. __Maintenance Plans__: Allow the definition and on-demand execution of plans with tasks that run inside the context of running pods. This allows for the definition of custom maintenance plans like backup and restore, data rebalance, etc.
+
+1. __Default Deploy Plan__: Built-in plan for deploying pods with both *serial* and *parallel* execution strategies.
+
+1. __Task FINISHED Goal State for Tasks within a Pod__: Support for tasks that run until finished. This is useful for operations such a formatting a data node volume.
+
+1. __Load Balanced Virtual Interface for Tasks__: Specify load balanced virtual interfaces for tasks. Virtual addresses define a name, source port, destination port mapping, protocol, and if it is externally advertised. Virtual addresses marked for external advertisement are discoverable via /endpoints HTTP API.
+
+1. __Placement Constraints within Pods__: Support for Marathon constrains language for pod (with the exception of the Marathon constrain language for volumes).
```
Comment (loren) [inline ROADMAP.md]: constraints

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+1. __Custom Strategies for Plans__: Java API with documentation for developing custom plan strategies.
+
+1. __DC/OS Networking API Integration for Tasks__: Support for integrating with DC/OS Networking. DC/OS supports [Container Networking Interface](https://github.com/containernetworking/cni/blob/master/SPEC.md).
+
+1. __Command Health Checks for Tasks__: Support for command health checks for tasks.
+
+1. __Basic Docker Image Support for Tasks__: Support for a single Docker image per pod.
+
+1. __Resource Sets__: Multiple different tasks can be run in sequence using the same resources (in particular persistent volumes) to produce a cumulative effect.  For example. initialization tasks before main tasks can be modeled in this way.
```
Comment (loren) [inline ROADMAP.md]: For example,

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+1. __Health Checks for Tasks__: Health check commands can be defined on a per Task basis to automatically kill unhealthy tasks.
```
Comment (loren) [inline ROADMAP.md]: Different than Command Health Checks for Tasks above?

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+1. __Updatable Placement Constraints for Pods__: Enable operators to update pods placement constraints so that pods can safely replace without losing data or violating performance SLAs.
+
+1. __Partition-Aware API Integration__: Support for Apache Mesos partition-aware frameworks APIs. For more details see [MESOS-5344](https://issues.apache.org/jira/browse/MESOS-5344) and [MESOS-6394](https://issues.apache.org/jira/browse/MESOS-6394).
+
+1. __Non-reserved Resources for Tasks__: Currently the SDK always reserves resources. This is because the initial focus for the SDK is stateful workloads, where reserving resources in essential for safe operations. For other workloads like twelve-factor apps and analytics jobs, reserving resources is generally undesirable.
+
+1. __Scale-in Count for Pods__: Reducing node (pod) count for most data services requires draining data. The SDK currently prevents reducing the pod count to avoid accidental deletion of data. With this feature, a plan could be called to drain data prior to termination. On termination, resources for the pods will be reserved. Note: scale-out, scale-up, scale-down are already supported.
```
Comment (loren) [inline ROADMAP.md]: On termination, resources for the pods will be unreserved?

**Comment by triclambert** [comment]
I would move CNI and containers to the Planned section.

File: ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+1. __Multiple Pod Types__: Support services with Heterogeneous nodes types - e.g., HDFS with Name Nodes, Journal Nodes, Data Nodes.
+
+1. __Configurable State Store Backing Service for Default Scheduler__: Enable configuring the state store backing service. Currently ZooKeeper is the only supported state store backing service.
+
+1. __Pod APIs__: Declarative pod specification via markup YAML API and Java API via the builder pattern.
+
+1. __Variable Substitution in YAML API__: Support variable substitution in YAML API - e.g., `type: {{TYPE}}`
+
+1. __Count for Pods__: Support for declaring the number of instances of a pod that should be running.
```
Comment (triclambert) [inline ROADMAP.md]: I think "count for pods" is a basic feature that can be assumed.

File: ROADMAP.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
+### Existing Features
+
+1. __Default Components__: Provide an extensible "default scheduler" to simplify building frameworks.
+
+1. __Multiple Pod Types__: Support services with Heterogeneous nodes types - e.g., HDFS with Name Nodes, Journal Nodes, Data Nodes.
```
Comment (triclambert) [inline ROADMAP.md]: Lowercase "heterogeneous".

File: ROADMAP.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+1. __Command Health Checks for Tasks__: Support for command health checks for tasks.
+
+1. __Basic Docker Image Support for Tasks__: Support for a single Docker image per pod.
+
+1. __Resource Sets__: Multiple different tasks can be run in sequence using the same resources (in particular persistent volumes) to produce a cumulative effect.  For example. initialization tasks before main tasks can be modeled in this way.
+
+1. __Health Checks for Tasks__: Health check commands can be defined on a per Task basis to automatically kill unhealthy tasks.
+
+1. __Readiness Checks for Tasks__: Add support for task readiness checks. Readiness checks enable configuration and software updates plans that restart pods to wait until data replication has caught up before proceeding.
```
Comment (triclambert) [inline ROADMAP.md]: s/Add support for task readiness checks.//

File: ROADMAP.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+### Planned Features
+
+1. __Advanced Docker Image Support for Tasks__: Support for Docker image per tasks.
```
Comment (triclambert) [inline ROADMAP.md]: s/tasks/task (within a pod)/

File: ROADMAP.md (hunk 1 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 2 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 3 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 4 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 5 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 6 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 7 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

File: ROADMAP.md (hunk 8 above)
Comment (keithchambers) [inline ROADMAP.md]: fixed

**Comment by keithchambers** [comment]
All comments have been fixed.

**Pull request merged**